src/input/gamepad/findGamepadInfo.js

```diff
@@ -0,0 +1,103 @@
+// @flow
+/*eslint indent:0*/
+
+import {gamepadInfoList} from "./gamepadInfoList";
+
+import type {GamepadID, GamepadInfo} from "./gamepadInfo";
+
+export function getGamepadNameAndInfo( identifier : string ) : null | [string, GamepadInfo] {
+
+  let name = null;
+  let info = null;
+  for (let i = 0; i < gamepadInfoList.length; i++) {
+    for (let j = 0; j < gamepadInfoList[i].ids.length; j++) {
+      if (checkAgainstGamepadID(identifier, gamepadInfoList[i].ids[j])) {
+        name = gamepadInfoList[i].ids[j].name;
+        info = gamepadInfoList[i];
+        break;
+      }
+    }
+  }
+
+  if (name === null || info === null) {
+    return null;
+  }
+  else {
+    return [name, info];
+  }
+
+}
+
+function checkAgainstGamepadID( identifier : string, gamepadID : GamepadID ) : bool {
+  let matchedVP = false;
+  if (gamepadID.vendor !== undefined && gamepadID.product !== undefined) {
+    const [gpdVendor, gpdProduct] = [gamepadID.vendor, gamepadID.product];
+    const vendorAndProduct = getVendorAndProduct(identifier);
+    if (vendorAndProduct !== null ) {
+      const [vendor, product] = vendorAndProduct;
+      matchedVP = match(vendor, gpdVendor) && match(product, gpdProduct);
+    }
+  }
+
+  let matchedID = false;
+  if (gamepadID.id !== undefined) {
+    const gpdID = gamepadID.id;
+    const id = getID(identifier);
+    const l = gpdID.length;
+    matchedID = gpdID.toLowerCase() === id.toLowerCase().substring(0,l);
+  }
+
+  return matchedVP || matchedID;
+}
+
+function removeZeroes (s : string) : string {
+  if (s.length < 2) {
+    return s;
+  }
+  else {
+    const [head, tail] = [s[0], s.substring(1)];
+    if (head === "0") {
+      return removeZeroes(tail);
+    }
+    else {
+      return s;
+    }
+  }
+}
+
+function match( s1 : string, s2 : string ) : bool {
+  return removeZeroes(s1).toLowerCase() === removeZeroes(s2).toLowerCase();  
+}
+
+
+// hacky functions to get name, vendor and product by munging strings
+
+function getVendorAndProduct( identifier : string ) : null | [string, string] {
+  const l = identifier.length;
+  let [vendor, product] = [null, null];
+  if (l > 9 && identifier[4] === "-" && identifier[9] === "-") {
```

Comment:
In Firefox the controller ID starts like "xxxx-yyyy-name" where "xxxx" is the vendor number and "yyyy" the product number.
In Chrome the controller ID ends with " (Vendor: xxxx Product: yyyy)".

These functions are just a hacky way to retrive the vendor/product from both these kinds of IDs.